state/redis/redis.go

```diff
@@ -52,34 +55,77 @@ func NewRedisStateStore(logger logger.Logger) *StateStore {
 	}
 }
 
-// Init does metadata and connection parsing
-func (r *StateStore) Init(metadata state.Metadata) error {
-	rand.Seed(time.Now().Unix())
+func parseRedisMetadata(meta state.Metadata) (metadata, error) {
+	m := metadata{}
 
-	connInfo := metadata.Properties
-	b, err := json.Marshal(connInfo)
-	if err != nil {
-		return err
+	if val, ok := meta.Properties[host]; ok && val != "" {
+		m.host = val
+	} else {
+		return m, errors.New("redis store error: missing host address")
 	}
 
-	var redisCreds credentials
-	err = json.Unmarshal(b, &redisCreds)
-	if err != nil {
-		return err
+	if val, ok := meta.Properties[password]; ok && val != "" {
+		m.password = val
 	}
 
-	ctx := context.Background()
-	opts := redisconn.Opts{
-		DB:       0,
-		Password: redisCreds.Password,
+	m.enableTLS = false
+	if val, ok := meta.Properties[enableTLS]; ok && val != "" {
+		tls, err := strconv.ParseBool(val)
+		if err != nil {
+			return m, fmt.Errorf("redis store error: can't parse enableTLS field: %s", err)
+		}
+		m.enableTLS = tls
+	}
+
+	m.maxRetries = 3
```

Comment:
Please create const DefaultMaxRetries = 3 , and set m.maxRetries = DefaultMaxRetries ..